mixnet/node/src/lib.rs

```diff
@@ -78,28 +82,36 @@ impl MixnetNode {
 
     async fn handle_connection(
         mut socket: TcpStream,
+        cache: ConnectionPool,
         private_key: PrivateKey,
         client_tx: mpsc::Sender<Body>,
     ) -> Result<(), Box<dyn Error>> {
         let body = Body::read(&mut socket).await?;
         match body {
-            Body::SphinxPacket(packet) => Self::handle_sphinx_packet(private_key, packet).await,
+            Body::SphinxPacket(packet) => {
+                cache.insert(
+                    socket.peer_addr().unwrap(),
+                    Arc::new(tokio::sync::Mutex::new(socket)),
+                );
+                Self::handle_sphinx_packet(cache, private_key, packet).await
```

Comment:
Ah, do we also need to add inbound connections to the map? I thought about keeping only outbound connections, but let me think.

Reply:
I think, even if we put inbound connections to the map, we won't use them again because the mixnet topology is static now (In case when we make the mixnet topology dynamic in the future, we will probably clear the map). Also, mixnodes won't send something to mixclient actively.

If we don't store inbound connections to the map, the connections are dropped and clients cannot reuse those connections anymore? Then, it makes sense.

Reply:
Yeah, if we drop on the server side, the client side is no longer valid.

Reply:
> the server-side may need to read these inbound connections periodically. The simplest solution would be have an infinite loop(task) for each inbound connection instead of putting them into the map. Hm, it seems this topic is more complicated than I expected. Let me run some tests and think about it again.

I totally agree, in this case, we need to periodically read inbound connections, once we accept a conn, we need to spawn a new thread to handle the logic.

Reply:
Yes. In my opinion, it's better to make the task continuously read the socket (then we don't need to store the socket in the map). From your implementation, it seems the connection is stored in the map and never be read anymore.

Reply:
For example, like 501b263 (I didn't make tests work).

Reply:
Yeah, makes sense. We do not need to store conn in the map anymore.